relay-server/src/utils/tracked_future.rs

```diff
@@ -0,0 +1,71 @@
+use futures::{sync::mpsc::Sender, Async, Future, Poll};
+
+/// Message send on the notification channel when the tracked future finishes or is disposed.
+pub struct TrackedFutureFinished;
+
+pub trait IntoTracked<F>
+where
+    F: Future,
+{
+    fn to_tracked(self, notifier: Sender<TrackedFutureFinished>) -> TrackedFuture<F>;
```

Comment:
_nit_: Usually these methods would be called with a verb like `.track()`. I'd propose to rename.

Reply:
ok